app/api/api_v1/endpoints/report.py

```diff
@@ -0,0 +1,143 @@
+
+from utils import plant_protection, irrigations, fertilisation, harvests, work_book
+
+router = APIRouter()
+
+
+@router.get("/{report_id}")
+def get_by_id(
+        report_id: int,
+        current_user: User = Depends(deps.get_current_user),
+        db: Session = Depends(deps.get_db)
+):
+    """
+    Returns a report via ID.
+    """
+
+    report_db = crud.report.get(db=db, id=report_id)
+
+    if not report_db:
+        raise HTTPException(
+            status_code=400,
+            detail="Report with ID:{} does not exist.".format(report_id)
+        )
+
+    types = {"work_book": "work-book", "plant_protection": "plant-protection", "irrigation": "irrigations", "fertilisations": "fertilisations", "harvest": "harvests", "global_gap": "GlobalGAP"}
+
+    if report_db.type == "work-book":
+        json_file = json.loads(report_db.file)
+
+        pdf = utils.work_book(
+            farm=utils.parse_farm_profile(json_file),
+            plot=utils.parse_plot_detail(json_file),
+            cult=utils.parse_generic_cultivation_info(json_file),
+            irri=utils.parse_irrigation(json_file),
+            fert=utils.parse_fertilization(json_file),
+            pdmd=utils.parse_plant_protection(json_file)
+        )
+
+    elif report_db.type == "plant-protection":
+
+        pdf = plant_protection(utils.parse_plant_protection(json.loads(report_db.file)))
+
+    elif report_db.type == "irrigations":
+
+        pdf = irrigations(utils.parse_irrigation(json.loads(report_db.file)))
+
+    elif report_db.type == "fertilisations":
+
+        pdf = fertilisation(utils.parse_fertilization(json.loads(report_db.file)))
+
+    elif report_db.type == "harvests":
+        # This will pass always, because it generates an empty .pdf every time
+        pdf = harvests()
+    else:
+        # Same as work-book for now
+        json_file = json.loads(report_db.file)
+
+        pdf = utils.work_book(
+            farm=utils.parse_farm_profile(json_file),
+            plot=utils.parse_plot_detail(json_file),
+            cult=utils.parse_generic_cultivation_info(json_file),
+            irri=utils.parse_irrigation(json_file),
+            fert=utils.parse_fertilization(json_file),
+            pdmd=utils.parse_plant_protection(json_file)
+        )
+
+    # Return the report as a response (binary)
+    headers = {
+        "Content-Disposition": "attachment; filename={}".format(report_db.name)
+    }
+
+    return Response(content=bytes(pdf.output()), media_type="application/pdf", headers=headers)
+
+
+@router.post("/{report_type}/dataset/{dataset_id}", response_model=ReportDBID)
+def create_by_data_id(
+        dataset_id: int,
+        report_type: str,
+        current_user: User = Depends(deps.get_current_user),
+        db: Session = Depends(deps.get_db)
+) -> ReportDBID:
+    """
+    Generate a report based off of a previously uploaded/queried data file.
```

Comment:
typo

Reply:
I'm not seeing it and my IDE isn't reporting it
I did update the text to better describe the api